tests/baselines/reference/keyofAndIndexedAccess2.errors.txt

```diff
@@ -83,6 +88,7 @@ tests/cases/conformance/types/keyof/keyofAndIndexedAccess2.ts(108,5): error TS23
         c[k] = 1; // Error, cannot write to index signature through constraint
         ~~~~
 !!! error TS2322: Type 'number' is not assignable to type 'T[keyof T]'.
+!!! error TS2322:   'number' is assignable to the constraint of type 'T[keyof T]', but 'T[keyof T]' could be instantiated with a different subtype of constraint 'number'.

```

Comment:
I guess this makes sense. I think our existing messages only talk about instantiations and constraints of type parameters directly, but in the compiler internals we talk about instantiations and constraints of structured types that _contain_ type parameters, like `T[keyof T]`. I just want to make sure @DanielRosenwasser feels good about promoting that concept to an actual error message since he suggested reusing this message. I was on the fence about whether this is an improvement at first, but after staring at it for a few minutes I don’t think I have an objection.

Reply:
As a user, it feels a bit weird to me to speak about the constraint of `T[keyof T]` like this - such a type, while indeed generic, wouldn’t be “constrained” in the same sense that `T extends U` is constrained by `U` (i.e. it doesn’t make sense to say here “solve for U”), so the term feels kind of nebulous in this context.

Reply:
We could switch it to "upper bound" or "minimal possible type" but if you're hitting this and you don't get it you're going to Google the message.

Reply:
I don’t have a suggestion better than “constraint”; I just think the added value is pretty marginal if you’re not already pretty expert.